### PR TITLE
MINIFICPP-2399 Evaluate expr language in GetFile's Directory property

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-2399

--------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/standard-processors/tests/unit/GetFileTests.cpp [hunk 1]
```diff
@@ -292,3 +292,26 @@ TEST_CASE("GetFile sets attributes correctly") {
...
+
+TEST_CASE("GetFile can use expression language in Directory property") {
+  using minifi::processors::GetFile;
+  LogTestController::getInstance().setTrace<GetFile>();
+
+  const auto get_file = std::make_shared<GetFile>("GetFile");
+  minifi::test::SingleProcessorTestController test_controller(get_file);
+
+  std::filesystem::path base_dir = test_controller.createTempDirectory();
+  auto date_str = date::format("%Y-%m-%d", std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()));
```
Comment (fgerlits) [inline extensions/standard-processors/tests/unit/GetFileTests.cpp:304]: nitpicking, but why do we cast (from microseconds?) to seconds if we only use the date part?

**Review by martinzink — APPROVED** [review]

File: extensions/standard-processors/tests/unit/GetFileTests.cpp (hunk 1 above)
Comment (lordgamez) [reply extensions/standard-processors/tests/unit/GetFileTests.cpp]: Good point, it's not needed here, updated in a9ebc13605cf3ac4ba35fd6816ee224f3395a9a9. The reason for this was that I started working with getDateTimeStr which takes `std::chrono::sys_seconds`, but after removing it's not needed anymore.

**Review by szaszm — APPROVED** [review]

**Review by fgerlits — APPROVED** [review]

File: extensions/standard-processors/tests/unit/GetFileTests.cpp [hunk 2]
```diff
@@ -292,3 +292,26 @@ TEST_CASE("GetFile sets attributes correctly") {
     CHECK(flow_file->getAttribute(minifi::core::SpecialFlowAttribute::FILENAME) == "beta.txt");
   }
 }
+
+TEST_CASE("GetFile can use expression language in Directory property") {
```
Comment (martinzink) [inline extensions/standard-processors/tests/unit/GetFileTests.cpp:296]: Based on the latest windows test failure you need something like here, and you would also need to define TZ_DATA_DIR in extensions/standard-processors/tests/CMakeLists.txt for unit tests aswell (currently its only defined for integration tests)
```suggestion
TEST_CASE("GetFile can use expression language in Directory property") {
#ifdef WIN32
  expression::dateSetInstall(TZ_DATA_DIR);
#endif
```

**Review by martinzink — CHANGES_REQUESTED** [review]

File: extensions/standard-processors/tests/unit/GetFileTests.cpp (hunk 2 above)
Comment (lordgamez) [reply extensions/standard-processors/tests/unit/GetFileTests.cpp:299]: Thanks! Updated in 70c7209706b5acda7b95051f57fad8168bdd7712

**Review by martinzink — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

**Pull request closed without merging**